File: python/leetcode020.py
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        map = {')':'(', '}':'{', ']':'['}
        ls = len(s)
        a = []
        i = 0
        while i < ls:
            if s[i] in ['(', '{', '[']:
                a.append(s[i])
            elif len(a) > 0 and map[s[i]] == a[-1]:
                a.pop()
            else:
                return False
            i = i + 1
        if len(a) == 0:
            return True
        else:
            return False

File: leetcodec++/leetcode020.cpp
//
// Created by 黄敏辉 on 2019-04-26.
//
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    bool isValid(string s) {

        stack<char> stack;
        for (int i = 0; i < s.length(); i++){
            if (s[i] == '{' || s[i] == '['||s[i] == '(')
                stack.push(s[i]);
            else if (!stack.empty()
                     && (
                             (s[i] == ')' && stack.top() == '(')
                             || (s[i] == '}' && stack.top() == '{')
                             || (s[i] == ']' && stack.top() == '[')
                     ))
                stack.pop();
            else
                return false;
        }
        return stack.empty();
    }
};